rust/operator-binary/src/superset_controller.rs

```diff
@@ -148,15 +175,59 @@ pub async fn reconcile_superset(
     for (rolegroup_name, rolegroup_config) in role_node_config.iter() {
         let rolegroup = superset.node_rolegroup_ref(rolegroup_name);
 
+        let authentication_classes: Vec<_> = superset
+            .spec
+            .authentication_config
+            .as_ref()
+            .iter()
+            .flat_map(|config| &config.methods)
+            .map(|method| &method.authentication_class)
+            .collect();
+
+        // async closures inside .map() are unstable... If they are stable we can use a simple authentication_classes.map(...)
+        let authentication_class = match authentication_classes.len() {
+            0 => None,
+            1 => {
+                let authentication_class_name = authentication_classes.get(0).unwrap();
+                Some(
+                    client
+                        .get::<AuthenticationClass>(authentication_class_name, None) // AuthenticationClass has ClusterScope
```

Comment:
Our `Controller` needs to have a watch on `AuthenticationClass` then, or we won't pick up changes to the `AuthenticationClass`.

Reply:
It now watches, thanks for the hint!